### ISSUE TITLE
import { JSEncrypt } from './JSEncrypt';

### ISSUE THREAD

**Issue by minikupa** [travist/jsencrypt#209]
D:\MK\project\node.js\covidcheck\node_modules\jsencrypt\lib\index.js:1
import { JSEncrypt } from './JSEncrypt';
^^^^^^

SyntaxError: Cannot use import statement outside a module

**Comment by izumiiAoba** [comment]
you can't use import statement (_supported by ES6_) in a node.js project
i also get stuck on this problem...

**Comment by pl4yradam** [comment]
+1

**Comment by travist** [comment]
Fixed in v3.1.0

**Comment by izumiiAoba** [comment]
i have updated to 3.1.0, but stiil got
```bash
import { JSEncrypt } from './JSEncrypt';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:1024:16)
    at Module._compile (node:internal/modules/cjs/loader:1072:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Module.require (node:internal/modules/cjs/loader:997:19)
    at require (node:internal/modules/cjs/helpers:92:18)
``` 
here is the statement i write to import
```js
const {JSEncrypt} = require('jsencrypt');
```

**Comment by liaohui5** [comment]
> i have updated to 3.1.0, but stiil got
> 
> ```shell
> import { JSEncrypt } from './JSEncrypt';
> ^^^^^^
> 
> SyntaxError: Cannot use import statement outside a module
>     at wrapSafe (node:internal/modules/cjs/loader:1024:16)
>     at Module._compile (node:internal/modules/cjs/loader:1072:27)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
>     at Module.load (node:internal/modules/cjs/loader:973:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
>     at Module.require (node:internal/modules/cjs/loader:997:19)
>     at require (node:internal/modules/cjs/helpers:92:18)
> ```
> 
> here is the statement i write to import
> 
> ```js
> const {JSEncrypt} = require('jsencrypt');
> ```

---
You can try this code

```
const { JSEncrypt } = require("jsencrypt/bin/jsencrypt");
```

**Comment by liaohui5** [comment]
> i have updated to 3.1.0, but stiil got
> 
> ```shell
> import { JSEncrypt } from './JSEncrypt';
> ^^^^^^
> 
> SyntaxError: Cannot use import statement outside a module
>     at wrapSafe (node:internal/modules/cjs/loader:1024:16)
>     at Module._compile (node:internal/modules/cjs/loader:1072:27)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
>     at Module.load (node:internal/modules/cjs/loader:973:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
>     at Module.require (node:internal/modules/cjs/loader:997:19)
>     at require (node:internal/modules/cjs/helpers:92:18)
> ```
> 
> here is the statement i write to import
> 
> ```js
> const {JSEncrypt} = require('jsencrypt');
> ```

You can try this code, Hopefully it helped you!

node.js:
```js
const JSEncrypt = require('JSEncrypt/bin/jsencrypt');
```
browser:
```js
import { JSEncrypt } from "jsencrypt/bin/jsencrypt";
```

**Comment by izumiiAoba** [comment]
> > i have updated to 3.1.0, but stiil got
> > ```shell
> > import { JSEncrypt } from './JSEncrypt';
> > ^^^^^^
> > 
> > SyntaxError: Cannot use import statement outside a module
> >     at wrapSafe (node:internal/modules/cjs/loader:1024:16)
> >     at Module._compile (node:internal/modules/cjs/loader:1072:27)
> >     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
> >     at Module.load (node:internal/modules/cjs/loader:973:32)
> >     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
> >     at Module.require (node:internal/modules/cjs/loader:997:19)
> >     at require (node:internal/modules/cjs/helpers:92:18)
> > ```
> > 
> > 
> > here is the statement i write to import
> > ```js
> > const {JSEncrypt} = require('jsencrypt');
> > ```
> 
> You can try this code, Hopefully it helped you!
> 
> node.js:
> 
> ```js
> const JSEncrypt = require('JSEncrypt/bin/jsencrypt');
> ```
> 
> browser:
> 
> ```js
> import { JSEncrypt } from "jsencrypt/bin/jsencrypt";
> ```

thank you, but i got another error
```js
const JSEncrypt = require('jsencrypt/bin/jsencrypt');
```
```bash
/... .../node_modules/jsencrypt/bin/jsencrypt.js:18
})(window, function() {
   ^

ReferenceError: window is not defined
    at Object.<anonymous> (/... .../node_modules/jsencrypt/bin/jsencrypt.js:18:4)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Module.require (node:internal/modules/cjs/loader:997:19)
    at require (node:internal/modules/cjs/helpers:92:18)
    at Object.<anonymous> (/... .../server/util/Cloud189.js:4:19)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
```

**Comment by liaohui5** [comment]
> > > i have updated to 3.1.0, but stiil got
> > > ```shell
> > > import { JSEncrypt } from './JSEncrypt';
> > > ^^^^^^
> > > 
> > > SyntaxError: Cannot use import statement outside a module
> > >     at wrapSafe (node:internal/modules/cjs/loader:1024:16)
> > >     at Module._compile (node:internal/modules/cjs/loader:1072:27)
> > >     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
> > >     at Module.load (node:internal/modules/cjs/loader:973:32)
> > >     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
> > >     at Module.require (node:internal/modules/cjs/loader:997:19)
> > >     at require (node:internal/modules/cjs/helpers:92:18)
> > > ```
> > > 
> > > 
> > > here is the statement i write to import
> > > ```js
> > > const {JSEncrypt} = require('jsencrypt');
> > > ```
> > 
> > 
> > You can try this code, Hopefully it helped you!
> > node.js:
> > ```js
> > const JSEncrypt = require('JSEncrypt/bin/jsencrypt');
> > ```
> > 
> > 
> > browser:
> > ```js
> > import { JSEncrypt } from "jsencrypt/bin/jsencrypt";
> > ```
> 
> thank you, but i got another error
> 
> ```js
> const JSEncrypt = require('jsencrypt/bin/jsencrypt');
> ```
> 
> ```shell
> /... .../node_modules/jsencrypt/bin/jsencrypt.js:18
> })(window, function() {
>    ^
> 
> ReferenceError: window is not defined
>     at Object.<anonymous> (/... .../node_modules/jsencrypt/bin/jsencrypt.js:18:4)
>     at Module._compile (node:internal/modules/cjs/loader:1108:14)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
>     at Module.load (node:internal/modules/cjs/loader:973:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
>     at Module.require (node:internal/modules/cjs/loader:997:19)
>     at require (node:internal/modules/cjs/helpers:92:18)
>     at Object.<anonymous> (/... .../server/util/Cloud189.js:4:19)
>     at Module._compile (node:internal/modules/cjs/loader:1108:14)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
> ```

You can refer to this issue: 
https://github.com/travist/jsencrypt/issues/147#issuecomment-534359519

**Comment by kg-currenxie** [comment]
But isn't the issue that the library is using `import` internally?

```
    ./node_modules/jsencrypt/lib/index.js:1
    import { JSEncrypt } from './JSEncrypt'; <-------- SyntaxError: Cannot use import statement outside a module
```

The node module should be built/compiled and provide a `/dist`, no? 🤔 
By default, Jest with babel is ignoring transpiling anything in node_modules

**Comment by minikupa** [comment]
> > i have updated to 3.1.0, but stiil got
> > ```shell
> > import { JSEncrypt } from './JSEncrypt';
> > ^^^^^^
> > 
> > SyntaxError: Cannot use import statement outside a module
> >     at wrapSafe (node:internal/modules/cjs/loader:1024:16)
> >     at Module._compile (node:internal/modules/cjs/loader:1072:27)
> >     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
> >     at Module.load (node:internal/modules/cjs/loader:973:32)
> >     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
> >     at Module.require (node:internal/modules/cjs/loader:997:19)
> >     at require (node:internal/modules/cjs/helpers:92:18)
> > ```
> > 
> > 
> > here is the statement i write to import
> > ```js
> > const {JSEncrypt} = require('jsencrypt');
> > ```
> 
> You can try this code, Hopefully it helped you!
> 
> node.js:
> 
> ```js
> const JSEncrypt = require('JSEncrypt/bin/jsencrypt');
> ```
> 
> browser:
> 
> ```js
> import { JSEncrypt } from "jsencrypt/bin/jsencrypt";
> ```

thank you! it works for me

**Comment by SilenceCreative** [comment]
now it's fix on version 3.3.1,
pnpm i jsencrypt@3.3.1
